### ISSUE TITLE
Add boolean data type

### ISSUE THREAD

**Issue by khushal-banks** [khushal-banks/logbank#15]
Add boolean data type for variable having only 1 or 0 as possible values

**Comment by khushal-banks** [comment]
A custom enum can be defined but would that be any better ?
- In terms of optimization - may be not
- In terms of readability - may be yes
- In terms of linking with other projects who has their own boolean implementation - It might just break things.

So, considering different options but any work on this should be done after #8 is either rejected or implemented.
- If #8 is implemented then let's go with a custom enum implementation which doesn't break things.
- If #8 is rejected then let's go with stdbool.h (standard is standard for a reason)

**Comment by jgabaut** [comment]
> C23 has removed boolean types [link](https://en.cppreference.com/w/c/types) so i will go with custom enum based definition

It seems `C23` removed `bool`, but not `_Bool`? [link](https://en.cppreference.com/w/c/keyword/_Bool)

If possible, I'd use that instead of rolling my own.

**Comment by khushal-banks** [comment]
> It seems `C23` removed `bool`, but not `_Bool`? [link](https://en.cppreference.com/w/c/keyword/_Bool)

Since C23 [they removed the need to include `stdbool.h`](https://en.cppreference.com/w/c/language/bool_constant) and predefined `bool`, `true` and `false` and removed from `stdbool.h`. New predefined `true`/`false` are defined as `non-lvalue`

So, my plan was:

```
#ifndef bool
typedef enum { false, true } bool;
#endif // bool
```

> If possible, I'd use that instead of rolling my own.

:+1: new plan:

```
#ifndef bool
#include <stdbool.h>
#endif // bool
```
